### DISCUSSION TITLE
Enhanced RTMP Using ffmpeg 6.1 ?

### DISCUSSION THREAD

**Discussion by chinbhat** [OvenMediaLabs/OvenMediaEngine#1480] [Help]
Hello Experts,

I hope one of you can help me with my usecase.

I am trying to stream H265 over SRT into the OME and looking to send H265 over RTMP to youtube. 

Now that ffmpeg 6.1 supports Enhanced RTMP to send H265 over RTMP youtube, is it possible to use the newer ffmpeg binary to accomplish this? Essentially, i am trying to cut down on the transcoding overhead to convert H265 to H264. Does the RTMP publish on OME support enhanced RTMP as well ? Is there a way for me to use the RTMP push of ffmpeg ?

I can do this using a standalong binary with ffmpeg, but i like the OME option of being able to provision using API's. 

Based on my testing, here is what my CPU usage looks like,

H265-SRT ingest - OME transcode to H264 and publish over RTMP. ~40% CPU Usage.
H265-SRT ingest - ffMPEG 6.1 stand alone, H265 over RTMP publish. ~ 2% CPU Usage

Appreciate your any help. thank you in adcance.

Cheers!!!

**Comment by getroot** [comment]
OME relies on ffmpeg's functionality for RTMP Push, so this shouldn't be difficult. However, as we are very busy and have many higher priority tasks, this may take a long time.

Enhanced RTMP is in our plans. https://github.com/AirenSoft/OvenMediaEngine/discussions/1167
However, we do not rely on ffmpeg for server functions such as RTMP Provider and write them all ourselves. This will probably be completed next year.
